File: tests/java/TestComposite.java
public class TestComposite {
  public static void main(String[] args) {
    System.out.println("Testing Java Composite...");
    try {
      Client.main(args);
    } catch (Exception e) {
      e.printStackTrace();
      System.exit(1);
    }
    System.out.println("Java Composite Test Passed!");
  }
}

File: tests/c/TestComposite.c
#define main pattern_main
#include "../../oop_in_c/Composite/Composite.c"
#undef main
#include <stdio.h>

int main() {
  printf("Testing C Composite...\n");
  // Call pattern main
  pattern_main();
  printf("C Composite Test Passed!\n");
  return 0;
}
